report/metadata_template.go

```diff
@@ -124,11 +124,14 @@ func (e MetadataTemplates) Copy() MetadataTemplates {
 // Merge merges two sets of MetadataTemplates so far just ignores based
 // on duplicate id key
 func (e MetadataTemplates) Merge(other MetadataTemplates) MetadataTemplates {
-	result := e.Copy()
+	if e == nil && other == nil {
+		return nil
+	}
+	result := make(MetadataTemplates, len(e))
```

Comment:
> Doesn't len(e)+len(other) assume there is no overlap?

Yes, so it will consume unnecessary memory if there is overlap but it will be optimal in terms of objects allocated (reducing garbage collector pressure).